_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -181,6 +181,7 @@ class FrameStorage
                 }
             }
         }
+        VP9_CHECK_AND_THROW((m_frameAllocator->DecreaseReference(frame.currFrame) == UMC::UMC_OK), MFX_ERR_UNKNOWN);
```

Comment:
please do not put function call to macro. Currently this code is ok, but what will happen, if somebody will change the macro? It could lead to double call of ```DecreaseReference``` and the issue would be hard to find. I suggest you to rewrite current code as following:
```
auto umc_sts = m_frameAllocator->DecreaseReference(frame.currFrame);
VP9_CHECK_AND_THROW(umc_sts == UMC::UMC_OK, MFX_ERR_UNKNOWN);
```

Reply:
@maksenox , could you please check  decoding with system memory, I have seen an problem on this use case when tried to move decrease references of decoded frame to synchronous part

Reply:
Yes, there is a problem, moving the reference counting to synchronous part cause different free frame order (the order in which player takes free surfaces and puts them into decoder). As the result, vaPutImage function receives different resolution of the source and destination frames. vaPutImage has a bug copying such frames and DRC streams on system memory on Linux may fail. That is just a coincidence that with this fix there are a little bit more failed tests.